crates/edr_provider/src/data.rs

```diff
@@ -546,9 +538,9 @@ impl ProviderData {
     /// Estimate the gas cost of a transaction. Matches Hardhat behavior.
     pub fn estimate_gas(
         &self,
-        transaction: PendingTransaction,
+        transaction: ExecutableTransaction,
         block_spec: &BlockSpec,
-    ) -> Result<u64, ProviderError> {
+    ) -> Result<Result<u64, EstimateGasFailure>, ProviderError> {
```

Comment:
I'm wondering if we could make `EstimateGasFailure` a variant of `ProviderError` instead of the nested results to simplify things?